internal/ethapi/api.go

```diff
@@ -1581,7 +1581,7 @@ func (s *PublicTransactionPoolAPI) GetRawTransactionByHash(ctx context.Context,
 func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
 	tx, blockHash, blockNumber, index, err := s.b.GetTransaction(ctx, hash)
 	if err != nil {
-		return nil, nil
+		return nil, err
```

Comment:
We return `nil` on purpose here. When the transaction doesn't exist, the return value of this RPC call should be JSON `null`.